### PR TITLE
add support for binding responsive styles in gen1 react sdk

### PR DESCRIPTION
**Pull request by steve8708**
currently gen2 SDKs support binding to responsive styles, like 

```js
{ bindings: { 'responsiveStyles.medium.color': 'state.foo'}
```

but not gen1, this fixes that

### CODE REVIEW

**Review by samijaber — COMMENTED** [review]
oh, this logic fixes responsive styles? how are they broken exactly? in SSR? 

You initially described this same code as solving SSR of styles, but when Clyde added tests it still didn't address that so he made a different change: https://github.com/BuilderIO/builder/pull/3563

**Comment by steve8708** [comment]
Where was SSR mentioned? This ( https://github.com/BuilderIO/builder/pull/3485 ) was the original PR, I didn't realize we removed this code that I added (was supposed to get in pre launch). The Pr title and description were

> 
> ​ add support for responsive styles in gen1 react sdk and memoize
> processBlock
> 
> 

> 
> ​
> 
> 

> 
> ​we don't use the post-processed responsiveStyles object that bindings
> influence, but instead the preprocessed one
> 
> 

> 
> ​
> 
> 

> 
> ​gen2 supports this, but not gen1
> 
> 

tl;dr you could not bind to responsiveStyles before, but now you can

**Comment by samijaber** [comment]
> maybe just miscommunication somewhere?

Seems to be the case. Sorry this didn't make it in sooner.

**Review by samijaber — APPROVED** [review]

**Comment by steve8708** [comment]
all good - best time to plant a tree is 10 years ago, second best time is today! thanks for the quick review
